src/presentation/controller/AbstractController.ts

```diff
@@ -0,0 +1,77 @@
+import { PageRouting, pageRoutingDefault, PagesRouting } from "./PageRouting";
+import PageType from "./PageType";
+
+abstract class AbstractController {
+  protected getRouting(
+    routing: PagesRouting,
+    pageType: PageType,
+    transactionId = "",
+    submissionId = ""
+  ) {
+    let pageRouting = routing.get(pageType);
+
+    if (!pageRouting) {
+      return pageRoutingDefault;
+    }
+
+    pageRouting = this.insertIdsInAllUrl(
+      pageRouting,
+      transactionId,
+      submissionId
+    );
+
+    return pageRouting;
+  }
+
+  protected pageType(path: string) {
+    const type = this.templateName(path);
+
+    return type as PageType;
+  }
+
+  protected templateName(url: string): string {
+    const splitted = url.split("/");
+    return splitted[splitted.length - 1];
+  }
+
+  insertIdsInUrl(url: string, transactionId = "", submissionId = ""): string {
+    url = this.insertSubmissionId(url, submissionId);
+    url = this.insertTransactionId(url, transactionId);
+    return url;
+  }
+
+  protected insertTransactionId(url: string, transactionId: string): string {
+    return transactionId ? url.replace(":transactionId", transactionId) : url;
+  }
+
+  protected insertSubmissionId(url: string, submissionId: string): string {
+    return submissionId ? url.replace(":submissionId", submissionId) : url;
+  }
+
+  protected insertIdsInAllUrl(
+    pageRouting: PageRouting,
+    transactionId: string,
+    submissionId: string
+  ): PageRouting {
+    return {
+      ...pageRouting,
+      previousUrl: this.insertIdsInUrl(
```

Comment:
you could create an inner function to do the inserting:
`const insertIds = (url: string) => this.insertIdsInUrl(url, transactionId, submissionId);`
then use it like:
`return {
      ...pageRouting,
    previousUrl: insertIds(pageRouting.previousUrl),`

Reply:
I think this comment was made before I updated this file